### ISSUE TITLE
GAN code is not compatible with the latest TensorFlow version

### ISSUE THREAD

**Issue by savan77** [adeshpande3/Tensorflow-Programs-and-Tutorials#1]
1 - Optimizer other than GradientDescentOptimizer will generate a ValueError. Can be solved by adding following line before optimizing. `with tf.variable_scope(tf.get_variable_scope(), reuse=False):`. There might be other solutions for this.

2 - Loss function takes named parameters. i.e `sigmoid_cross_entropy_with_logits(logits=Dx, labels=tf.ones_like(Dx)`

**Comment by adeshpande3** [comment]
Thanks for the catch. Just pushed the changes
